### PR TITLE
implements As Nature and God Intended.

### LINKED ISSUE #608: W2D - As Nature and God Intended

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes #608

### CODE REVIEW

**Comment by stopfstedt** [comment]
@mmeldo i have one question with this card, and one implementation problem.

_question:_ can this effect target kung-fu dudes and weapons that are not part of the shootout?  not sure what the scope here is exactly.

_problem:_ the effects only seem to be taking hold on weapons and dudes controlled by the player who played this event. i can't quite figure out how to target the opponent's cards. any suggestions?

**Comment by mmeldo** [comment]
> _question:_ can this effect target kung-fu dudes and weapons that are not part of the shootout?  not sure what the scope here is exactly.

I am almost 100% sure that shootout effects affect only dudes or goods attached to dudes participating in a shootout.

**Comment by mmeldo** [comment]
> _problem:_ the effects only seem to be taking hold on weapons and dudes controlled by the player who played this event. i can't quite figure out how to target the opponent's cards. any suggestions?

Use `targetController: 'any'` in the `lastingEffect`

**Comment by stopfstedt** [comment]
@mmeldo - i think this is _almost_ ready to go, the only thing that i can't quite figure out is how to turn off only 'shootout' abilities on attachments. `cannotTriggerCardAbilities()` seems to be the right method, i see the hook for passing in a callback as well, but i'm stuck at that point since i can't quite suss out which attribute of the effect to filter on. or perhaps this needs a dedicated method?

**Comment by mmeldo** [comment]
> @mmeldo - i think this is _almost_ ready to go, the only thing that i can't quite figure out is how to turn off only `shootout` abilities on attachments. `cannotTriggerCardAbilities()` seems to be the right method, i see the hook for passing in a callback as well, but i'm stuck at that point since i can't quite suss out which attribute of the effect to filter on. or perhaps this needs a dedicated method? as it stands, this implementation turns off _all_ card abilities on attachments that are in the fight, including cheating resolutions, which is obviously not correct.

You can pass a condition to the `cannotTriggerCardAbilities()` effect with `ability` parameter.
Then you can check if the ability is a shootout ability by checking if it has "shootout" in the `playType`. There can be abilities with "noon/shootout" but since the  `As Nature and God Intended` will last only in shootout we do not care. And I am not aware of any abilities that can be triggered in shootout plays and also in resolution.
```
ability => ability.playType && ability.playType.includes('shootout')
```
I did not test this, but I believe that should work.

**Comment by stopfstedt** [comment]
@mmeldo thanks for the feedback, i'll try to wrap this one up today and get started with the next one.

**Comment by stopfstedt** [comment]
> Edit: You can actually make sure you are currently in "shootout plays" but adding this condition:
> `ability.game.isShootoutPlayWindow()`

this didn't work, and doesn't seem to be necessary. worth a try though.

I found a different problem though, and that has to do with the card blanking effect of `blankExcludingKeywords`. this is too far- reaching, as this flags the attachment as (partially) blanked and therefore  blocks other, non-shootout abilities (like cheatin' resolutions) as well.
For example, this turns off Quickdraw Handgun's cheatin' resolution. 

What would be needed here is a more fine-grained approach to handling blanking effects/blank card states. I peeked at the relevant code and separating the blanking of traits from the blanking of abilities seems a bit out of scope for this card implementation/would require a more involved rewrite.

I'll file this as a separate issue and sit on this card for the time being.

**Comment by mmeldo** [comment]
Added new effects: `traitBlank` and `bulletBonusBlank`

**Comment by stopfstedt** [comment]
@mmeldo happy to report that the new blanking effects work like a charm. Click-tested  with a full assortment of weapon attachment - Shotgun, Flame-Thrower, Quickdraw Handgun, etc., this now works as intended.

If you have any thoughts on how to improve the message output for this action please let me know (or add one yourself), i left this as a TODO in there.

**Comment by mmeldo** [comment]
> If you have any thoughts on how to improve the message output for this action please let me know (or add one yourself), i left this as a TODO in there.

Well, I have 2 suggestions. One is verbose:
`{0} uses {1} to have all Weapons lose their traits, bullet bonuses and Shootout abilities, and make all Kung Fu dudes studs until end of this round`
and the other is not, but it is not that clear:
`{0} uses {1} to disable all Weapons and make all Kung Fu dudes studs until end of this round`

You can use any of these and I will be fine with that. Or neither and maybe you can come up with something better.

**Comment by stopfstedt** [comment]
i went with the second suggestion, thanks for that. ready for re-review.

File: server/game/cards/15-W2D/AsNatureAndGodIntended.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const ActionCard = require('../../actioncard.js');
+
+class AsNatureAndGodIntended extends ActionCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'As Nature and God Intended',
+            playType: ['shootout'],
+            message: context =>
+                this.game.addMessage(' {0} uses {1} to disable all Weapons and make all Kung Fu dudes studs until end of this round', context.player, this), // TODO: Not sure about what to output here, regurgitating the card's effect is too verbose. [ST 2021/07/07]
```
Comment (mmeldo) [inline server/game/cards/15-W2D/AsNatureAndGodIntended.js:9]: You can delete the TODO and I know I suggested it, but I believe the message should end: "... until end of this shootout round" to make clear it does not last until the end of the day (round).

**Review by mmeldo — CHANGES_REQUESTED** [review]

File: server/game/cards/15-W2D/AsNatureAndGodIntended.js (hunk 1 above)
Comment (stopfstedt) [reply server/game/cards/15-W2D/AsNatureAndGodIntended.js]: :heavy_check_mark:

**Comment by mmeldo** [comment]
Man, that was fast 🚄

**Review by mmeldo — APPROVED** [review]

**Pull request merged**